central/virtualmachine/service/service_impl.go

```diff
@@ -65,8 +72,18 @@ func (s *serviceImpl) GetVirtualMachine(ctx context.Context, request *v2.GetVirt
 }
 
 func (s *serviceImpl) ListVirtualMachines(ctx context.Context, request *v2.ListVirtualMachinesRequest) (*v2.ListVirtualMachinesResponse, error) {
-	// TODO: Filtering/search capabilities
-	vms, err := s.datastore.GetAllVirtualMachines(ctx)
+	searchQuery := search.EmptyQuery()
+	requestQuery := request.GetQuery().GetQuery()
+	if requestQuery != "" {
+		parsedQuery, err := search.ParseQuery(request.GetQuery().GetQuery())
+		if err != nil {
+			return nil, errors.Wrap(err, "parsing input query")
+		}
+		searchQuery = parsedQuery
+	}
+	paginated.FillPaginationV2(searchQuery, request.GetQuery().GetPagination(), defaultPageSize)
```

Comment:
What is the default sorting pattern here? Perhaps sorting the response by name + namespace would make sense?